unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/ParcelScenesCleaner.cs

```diff
@@ -92,94 +117,58 @@ public void ForceCleanup()
 
                 if (!scenesToRemove.Contains(markedEntityInfo.scene))
                     scenesToRemove.Add(markedEntityInfo.scene);
+
+                if (!immediate && DCLTime.realtimeSinceStartup - lastTime >= MAX_TIME_BUDGET)
+                {
+                    yield return null;
+                    lastTime = Time.unscaledTime;
+                }
             }
 
             while (entitiesMarkedForCleanup.Count > 0)
             {
                 IDCLEntity entity = entitiesMarkedForCleanup.Dequeue();
                 entity.SetParent(null);
                 entity.Cleanup();
+
+                if (!immediate && DCLTime.realtimeSinceStartup - lastTime >= MAX_TIME_BUDGET)
+                {
+                    yield return null;
+                    lastTime = Time.unscaledTime;
```

Comment:
Shouldn't this be also DCLTime.realtimeSinceStartup?

Reply:
`DCLTime.realtimeSinceStartup` is less performant than `Time.unscaledTime`, but it works for cases where we need to measure time between instructions without skipping a frame. This is because `Time.time` and `Time.unscaledTime` is just updated a single time when the frame starts. Here, as we just skipped a frame, `Time.unscaledTime` is enough.